### PR TITLE
chore: roll driver to 1.45.0 beta, implement new features

### PR DESCRIPTION
**Pull request by yury-s**
Reference: https://github.com/microsoft/playwright-java/issues/1586

### CODE REVIEW

File: playwright/src/main/java/com/microsoft/playwright/impl/APIRequestImpl.java [hunk 1]
```diff
@@ -58,6 +58,7 @@ private APIRequestContextImpl newContextImpl(NewContextOptions options) {
       options.storageState = null;
     }
     JsonObject params = gson().toJsonTree(options).getAsJsonObject();
+    System.out.println(params.toString());
```
Comment (dgozman) [inline playwright/src/main/java/com/microsoft/playwright/impl/APIRequestImpl.java:61]: Remove?

File: playwright/src/main/java/com/microsoft/playwright/impl/ClockImpl.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    JsonObject params = new JsonObject();
+    params.addProperty("timeNumber", time.getTime());
+    browserContext.sendMessage("clockSetSystemTime", params);
+  }
+
+  private static void parseTime(Object time, JsonObject params) {
+    if (time instanceof Long) {
+      params.addProperty("timeNumber", (Long) time);
+    } else if (time instanceof Date) {
+      params.addProperty("timeString", ((Date) time).getTime());
```
Comment (dgozman) [inline playwright/src/main/java/com/microsoft/playwright/impl/ClockImpl.java:124]:
```suggestion
      params.addProperty("timeNumber", ((Date) time).getTime());
```

File: playwright/src/test/java/com/microsoft/playwright/TestPageCheck.java [hunk 3]
```diff
@@ -32,6 +32,7 @@ void shouldCheckTheLabelWithPosition() {
     BoundingBox box = page.querySelector("text=Click me").boundingBox();
     page.check("text=Click me", new Page.CheckOptions().setPosition(box.width - 10, 2));
     assertEquals(true, page.evalOnSelector("input", "input => input.checked"));
+    page.unrouteAll();
```
Comment (dgozman) [inline playwright/src/test/java/com/microsoft/playwright/TestPageCheck.java:35]: Unrelated?

File: playwright/src/main/java/com/microsoft/playwright/impl/Utils.java [hunk 4]
```diff
@@ -173,40 +175,104 @@ static String mimeType(Path path) {
...
-        }
-        jsonStreams.add(temp.toProtocolRef());
+      if (localDirectory != null) {
+        localPaths = collectFiles(localDirectory);
+      }
+      JsonObject json = createTempFiles(context, localDirectory, localPaths);
+      JsonArray writableStreams = json.getAsJsonArray("writableStreams");
+      JsonArray jsonStreams = copyLocalToTempFiles(context, localPaths, writableStreams);
+      if (json.has("rootDir")) {
+        params.add("directoryStream", json.get("rootDir"));
```
Comment (dgozman) [inline playwright/src/main/java/com/microsoft/playwright/impl/Utils.java:192]: When passing `directoryStream`, we should not set `streams` property.

**Review by dgozman — APPROVED** [review]

File: playwright/src/main/java/com/microsoft/playwright/impl/ClockImpl.java (hunk 2 above)
Comment (yury-s) [reply playwright/src/main/java/com/microsoft/playwright/impl/ClockImpl.java:124]: Somehow the tests passed!
